aioxmpp/avatar/service.py

```diff
@@ -509,19 +566,20 @@ def publish_avatar_set(self, avatar_set):
 
         id_ = avatar_set.png_id
 
-        yield from self._pubsub.publish(
-            None,
-            namespaces.xep0084_data,
-            avatar_xso.Data(avatar_set.image_bytes),
-            id_=id_
-        )
+        with (yield from self._lock):
```

Comment:
Actually, I wonder whether we already get a push notification with the avatar data or at least the item ID because PEP. Can you investigate that? Does that even help us? (I’m not sure about that.)

Downloading the full avatar does seem like not a very bad thing to do here, unless we find a saner way. Matching the metadata won’t be sufficient to detect more than half of the collision scenarios.